imap_processing/codice/codice_l1a_lo_priority.py

```diff
@@ -0,0 +1,256 @@
+        sensor=view_tab_info["sensor"],
+        three_d_collapsed=view_tab_info["3d_collapse"],
+        collapse_table=view_tab_info["collapse_table"],
+    )
+
+    if view_tab_obj.sensor != 0:
+        raise ValueError("Unsupported sensor ID for Lo priority processing.")
+
+    # ========== Get Voltage Data from LUT ===========
+    # Use plan id and plan step to get voltage data's table_number in ESA sweep table.
+    # Voltage data is (128,)
+    esa_table_number = sci_lut_data["plan_tab"][f"({plan_id}, {plan_step})"][
+        "lo_stepping"
+    ]
+    voltage_data = sci_lut_data["esa_sweep_tab"][f"{esa_table_number}"]
+
+    # ========= Get Epoch Time Data ===========
+    # Epoch center time and delta
+    epoch_center, deltas = get_codice_epoch_time(
+        unpacked_dataset["acq_start_seconds"].values,
+        unpacked_dataset["acq_start_subseconds"].values,
+        unpacked_dataset["spin_period"].values,
+        view_tab_obj,
+    )
+
+    # ========= Decompress and Calculate Reshape information ===========
+    # Set needed metadata for Hi and Lo's different priority products
+    if apid == CODICEAPID.COD_LO_SW_PRIORITY_COUNTS:
+        species_names = sci_lut_data["data_product_lo_tab"]["0"]["priority"]["sw"][
+            "species_names"
+        ]
+        logical_source_id = "imap_codice_l1a_lo-sw-priority"
+        compression_algorithm = constants.LO_COMPRESSION_ID_LOOKUP[view_tab_obj.view_id]
+    elif apid == CODICEAPID.COD_LO_NSW_PRIORITY_COUNTS:
+        species_names = sci_lut_data["data_product_lo_tab"]["0"]["priority"]["nsw"][
+            "species_names"
+        ]
+        logical_source_id = "imap_codice_l1a_lo-nsw-priority"
+        compression_algorithm = constants.LO_COMPRESSION_ID_LOOKUP[view_tab_obj.view_id]
+    else:
+        raise ValueError("Unsupported APID for Lo priority processing.")
+
+    # Decompress data using byte count information from decommed data
+    binary_data_list = unpacked_dataset["data"].values
+    byte_count_list = unpacked_dataset["byte_count"].values
+
+    # The decompressed data in the shape of (epoch, n). Then reshape later.
+    decompressed_data = [
+        decompress(
+            packet_data[:byte_count],
+            compression_algorithm,
+        )
+        for (packet_data, byte_count) in zip(
+            binary_data_list, byte_count_list, strict=False
+        )
+    ]
+
+    num_packets = len(binary_data_list)
+
+    # Reshape decompressed data to in below for loop:
+    # (num_packets, num_species, esa_steps, collapse_shape[0](spin_sector))
+    num_species = len(species_names)
+    esa_steps = constants.NUM_ESA_STEPS
+    collapse_shape = get_collapse_pattern_shape(
+        sci_lut_data,
+        view_tab_obj.sensor,
+        view_tab_obj.collapse_table,
+    )
+    print(f"Collapse shape: {collapse_shape} - num_species: {num_species}")
+    print(f"decompressed data {np.array(decompressed_data).shape}")
+    species_data = np.array(decompressed_data).reshape(
```

Comment:
All binary data arrays should get a `uint8` or similar dtype when you are calling np.array() on them so that they don't go to int64 by default.